File: targets/csharp-aspnet/Controllers/ProfileController.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;
using Microsoft.AspNetCore.Mvc;

namespace BadDemo.Controllers
{
    [ApiController]
    [Route("profile")]
    public class ProfileController : ControllerBase
    {
        [Serializable]
        public class User
        {
            public string Username = "";
            public string Email = "";
            public string Role = "user";
        }

        /// <summary>Profile update request data.</summary>
        public class UpdateProfileRequest
        {
            public string Username { get; set; } = "";
            public string Email { get; set; } = "";
            public string Role { get; set; } = "user";
        }

        private static readonly Dictionary<string, User> Users = new Dictionary<string, User>();

        [HttpPost("update")]
        public IActionResult UpdateProfile([FromBody] UpdateProfileRequest req)
        {
            if (!Users.TryGetValue(req.Username, out var user))
            {
                user = new User();
                Users[req.Username] = user;
            }
            user.Email = req.Email;
            user.Role = req.Role;
            return Ok(user);
        }

        [HttpPost("import")]
        public IActionResult ImportProfile([FromBody] string b64)
        {
            byte[] data = Convert.FromBase64String(b64);
            var formatter = new BinaryFormatter();
            using var ms = new MemoryStream(data);
            var user = (User)formatter.Deserialize(ms);
            Users[user.Username] = user;
            return Ok("imported " + user.Username);
        }
    }
}

File: targets/php-laravel/app/Http/Controllers/ProfileController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;

class ProfileUser
{
    public string $username = '';
    public string $email = '';
    public string $role = 'user';
}

class ProfileController
{
    private static array $users = [];

    private function userStore(string $username): ProfileUser
    {
        if (!isset(self::$users[$username])) {
            self::$users[$username] = new ProfileUser();
        }
        return self::$users[$username];
    }

    public function update(Request $request)
    {
        $username = $request->input('username', '');
        $user = $this->userStore($username);
        foreach ($request->all() as $key => $value) {
            $user->$key = $value;
        }
        return response()->json([
            'username' => $user->username,
            'email' => $user->email,
            'role' => $user->role,
        ]);
    }

    public function import(Request $request)
    {
        $data = $request->getContent();
        $user = unserialize($data);
        self::$users[$user->username] = $user;
        return response()->json(['imported' => $user->username]);
    }
}
